src/commands/keygen.rs

```diff
@@ -0,0 +1,605 @@
+
+                            _ => {
+                                return Err(format!("Invalid value {s:?} for algorithm (-a)"));
+                            }
+                        })
+                    })?;
+                }
+
+                Arg::Short('k') => {
+                    // NOTE: '-k' can be repeated, to no effect.
+                    make_ksk = true;
+                }
+
+                Arg::Short('b') => {
+                    // NOTE: '-b' can be repeated; the last instance wins.
+                    bits = parse_os("bits (-b)", &parser.value()?)?;
+                }
+
+                Arg::Short('r') => {
+                    // NOTE: '-r' can be repeated; we don't use it, so the order doesn't matter.
+                    let _ = parser.value()?;
+                }
+
+                Arg::Short('s') => {
+                    // NOTE: '-s' can be repeated, to no effect.
+                    create_symlinks = true;
+                }
+
+                Arg::Short('f') => {
+                    // NOTE: '-f' can be repeated, to no effect.
+                    force_symlinks = true;
+                }
+
+                // TODO: '-v' version argument?
+                Arg::Value(value) => {
+                    if name.is_some() {
+                        return Err("cannot specify multiple domain names".into());
+                    }
+
+                    name = Some(parse_os("domain name", &value)?);
+                }
+
+                Arg::Short(x) => return Err(format!("Invalid short option: -{x}").into()),
+                Arg::Long(x) => {
+                    return Err(format!("Long options are not supported, but `--{x}` given").into())
+                }
+            }
+        }
+
+        let algorithm = match algorithm {
+            Some(SecAlg::RSASHA256) => GenerateParams::RsaSha256 { bits },
+            Some(SecAlg::ECDSAP256SHA256) => GenerateParams::EcdsaP256Sha256,
+            Some(SecAlg::ECDSAP384SHA384) => GenerateParams::EcdsaP384Sha384,
+            Some(SecAlg::ED25519) => GenerateParams::Ed25519,
+            Some(SecAlg::ED448) => GenerateParams::Ed448,
+            Some(_) => unreachable!(),
+            None => {
+                return Err("Missing algorithm (-a) option".into());
+            }
+        };
+
+        let symlink = match (create_symlinks, force_symlinks) {
+            (true, true) => SymlinkArg::Force,
+            (true, false) => SymlinkArg::Yes,
+            // If only '-f' is specified, no symlinking is done.
+            (false, _) => SymlinkArg::No,
+        };
+
+        let Some(name) = name else {
+            return Err("Missing domain name argument".into());
+        };
+
+        Ok(Self {
+            algorithm,
+            make_ksk,
+            symlink,
+            name,
+        })
+    }
+}
+
+impl From<Keygen> for Command {
+    fn from(value: Keygen) -> Self {
+        Self::Keygen(value)
+    }
+}
```

Comment:
I put all of these in `commands/mod.rs`, though maybe you're right it should be here.

Reply:
I'm mostly opposed to the inconsistency

Reply:
I think we should move each one to its respective module.  I don't know whether I should try moving them all to their modules in this PR, or to do all of them later.